{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "V5E1"
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "TPU"
  },
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jIY5FzICUNUB",
        "outputId": "11d252cd-e5d5-47d7-8914-057795ed9831"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rHit:1 https://cli.github.com/packages stable InRelease\n",
            "Get:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n",
            "Hit:3 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Get:5 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Get:7 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,437 kB]\n",
            "Get:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease [18.1 kB]\n",
            "Get:9 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [3,526 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:11 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,826 kB]\n",
            "Hit:12 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:13 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 Packages [38.5 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,856 kB]\n",
            "Fetched 20.1 MB in 2s (10.0 MB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "build-essential is already the newest version (12.9ubuntu3).\n",
            "unzip is already the newest version (6.0-26ubuntu3.2).\n",
            "The following additional packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "Suggested packages:\n",
            "  blends-doc menu-l10n gksu | kde-runtime | ktsuss\n",
            "The following NEW packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "  sra-toolkit\n",
            "0 upgraded, 7 newly installed, 0 to remove and 4 not upgraded.\n",
            "Need to get 8,290 kB of archives.\n",
            "After this operation, 23.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n",
            "Fetched 8,290 kB in 2s (4,210 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package menu.\n",
            "(Reading database ... 117072 files and directories currently installed.)\n",
            "Preparing to unpack .../0-menu_2.1.47ubuntu4_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu4) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../1-blends-common_0.7.4ubuntu1_all.deb ...\n",
            "Unpacking blends-common (0.7.4ubuntu1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../2-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../3-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../4-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../5-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../6-med-config_3.7.1_all.deb ...\n",
            "Unpacking med-config (3.7.1) ...\n",
            "Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up menu (2.1.47ubuntu4) ...\n",
            "Setting up blends-common (0.7.4ubuntu1) ...\n",
            "Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Setting up med-config (3.7.1) ...\n",
            "Adding group `med' (GID 107) ...\n",
            "Done.\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.11) ...\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "--2025-11-11 05:09:28--  https://github.com/alexdobin/STAR/archive/2.7.10a.zip\n",
            "Resolving github.com (github.com)... 140.82.116.3\n",
            "Connecting to github.com (github.com)|140.82.116.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://codeload.github.com/alexdobin/STAR/zip/refs/tags/2.7.10a [following]\n",
            "--2025-11-11 05:09:28--  https://codeload.github.com/alexdobin/STAR/zip/refs/tags/2.7.10a\n",
            "Resolving codeload.github.com (codeload.github.com)... 140.82.116.9\n",
            "Connecting to codeload.github.com (codeload.github.com)|140.82.116.9|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: unspecified [application/zip]\n",
            "Saving to: ‘2.7.10a.zip’\n",
            "\n",
            "2.7.10a.zip             [      <=>           ]  11.86M  6.48MB/s    in 1.8s    \n",
            "\n",
            "2025-11-11 05:09:30 (6.48 MB/s) - ‘2.7.10a.zip’ saved [12437043]\n",
            "\n",
            "Archive:  2.7.10a.zip\n",
            "7f0f3f16ebf232963e99891b5d3fecba071bdd57\n",
            "   creating: STAR-2.7.10a/\n",
            "  inflating: STAR-2.7.10a/.gitignore  \n",
            " extracting: STAR-2.7.10a/.gitmodules  \n",
            "  inflating: STAR-2.7.10a/.travis.yml  \n",
            "  inflating: STAR-2.7.10a/CHANGES.md  \n",
            "  inflating: STAR-2.7.10a/CODE_OF_CONDUCT.md  \n",
            "  inflating: STAR-2.7.10a/CONTRIBUTING.md  \n",
            "  inflating: STAR-2.7.10a/LICENSE    \n",
            "  inflating: STAR-2.7.10a/README.md  \n",
            "  inflating: STAR-2.7.10a/RELEASEnotes.md  \n",
            "  inflating: STAR-2.7.10a/_config.yml  \n",
            "   creating: STAR-2.7.10a/bin/\n",
            "   creating: STAR-2.7.10a/bin/Linux_x86_64/\n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64/STAR  \n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64/STARlong  \n",
            "   creating: STAR-2.7.10a/bin/Linux_x86_64_static/\n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64_static/STAR  \n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64_static/STARlong  \n",
            "   creating: STAR-2.7.10a/bin/MacOSX_x86_64/\n",
            "  inflating: STAR-2.7.10a/bin/MacOSX_x86_64/STAR  \n",
            "  inflating: STAR-2.7.10a/bin/MacOSX_x86_64/STARlong  \n",
            "   creating: STAR-2.7.10a/doc/\n",
            "  inflating: STAR-2.7.10a/doc/STARmanual.pdf  \n",
            "   creating: STAR-2.7.10a/docs/\n",
            "  inflating: STAR-2.7.10a/docs/STARconsensus.md  \n",
            "  inflating: STAR-2.7.10a/docs/STARsolo.md  \n",
            "   creating: STAR-2.7.10a/extras/\n",
            "   creating: STAR-2.7.10a/extras/doc-latex/\n",
            "  inflating: STAR-2.7.10a/extras/doc-latex/STARmanual.tex  \n",
            "  inflating: STAR-2.7.10a/extras/doc-latex/convertParDefToLatexTable.awk  \n",
            "  inflating: STAR-2.7.10a/extras/doc-latex/parametersDefault.tex  \n",
            "   creating: STAR-2.7.10a/extras/docker/\n",
            "  inflating: STAR-2.7.10a/extras/docker/Dockerfile  \n",
            "   creating: STAR-2.7.10a/extras/parameters/\n",
            "  inflating: STAR-2.7.10a/extras/parameters/ENCODE.txt  \n",
            "   creating: STAR-2.7.10a/extras/scripts/\n",
            "  inflating: STAR-2.7.10a/extras/scripts/calcInsertCoverage.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/extractSJfromGTF.sh  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/filterCirc.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/mergeGeneCounts.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/mergeLogFinal.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/mergeSuperContig.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjBED12.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjCollapseSamples.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjFromSAMcollapseUandM.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjFromSAMcollapseUandM_inclOverlaps.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjMotif.m  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloBasicCellFilter.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloCompareMtx.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloCountMatrixFromBAM.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloExtractFiltCells.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloUMIperCell.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/tagXSstrandedData.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/transcriptTypes.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/transfragsFromBedGraph.awk  \n",
            "   creating: STAR-2.7.10a/extras/tests/\n",
            "   creating: STAR-2.7.10a/extras/tests/scripts/\n",
            "  inflating: STAR-2.7.10a/extras/tests/scripts/checkCellReadsStats.awk  \n",
            "  inflating: STAR-2.7.10a/extras/tests/scripts/checkCellReadsStats_vsBAM.awk  \n",
            "  inflating: STAR-2.7.10a/extras/tests/scripts/checkCellReadsStats_vsMatrix.awk  \n",
            "   creating: STAR-2.7.10a/source/\n",
            "  inflating: STAR-2.7.10a/source/AlignVsTranscript.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortByCoordinate.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortByCoordinate.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortUnmapped.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortUnmapped.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMfunctions.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMfunctions.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMoutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMoutput.h  \n",
            "  inflating: STAR-2.7.10a/source/Chain.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Chain.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign_chimericBAMoutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign_chimericJunctionOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign_chimericStitching.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericDetection.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericDetection.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericDetection_chimericDetectionMult.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericSegment.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericSegment.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericTranscript.h  \n",
            "  inflating: STAR-2.7.10a/source/ClipCR4.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ClipCR4.h  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate.h  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate_clip.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate_clipChunk.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate_initialize.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ErrorWarning.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ErrorWarning.h  \n",
            "  inflating: STAR-2.7.10a/source/GTF.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GTF.h  \n",
            "  inflating: STAR-2.7.10a/source/GTF_superTranscript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GTF_transcriptGeneSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome.h  \n",
            "  inflating: STAR-2.7.10a/source/Genome_genomeGenerate.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_genomeLoad.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_genomeOutLoad.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_insertSequences.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_transformGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GlobalVariables.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GlobalVariables.h  \n",
            "  inflating: STAR-2.7.10a/source/InOutStreams.cpp  \n",
            "  inflating: STAR-2.7.10a/source/InOutStreams.h  \n",
            "  inflating: STAR-2.7.10a/source/IncludeDefine.h  \n",
            "  inflating: STAR-2.7.10a/source/Makefile  \n",
            "  inflating: STAR-2.7.10a/source/OutSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/OutSJ.h  \n",
            "  inflating: STAR-2.7.10a/source/PackedArray.cpp  \n",
            "  inflating: STAR-2.7.10a/source/PackedArray.h  \n",
            "  inflating: STAR-2.7.10a/source/ParameterInfo.h  \n",
            "  inflating: STAR-2.7.10a/source/Parameters.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersChimeric.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersChimeric_initialize.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersClip.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersClip_initialize.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersGenome.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersSolo.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersSolo.h  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_closeReadsFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_openReadsFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_readFilesInit.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_readSAMheader.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_samAttributes.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Quantifications.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Quantifications.h  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign.h  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk.h  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk_mapChunk.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk_processChunks.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_CIGAR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_alignBAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_assignAlignToWindow.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_calcCIGAR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetection.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetectionOld.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetectionOldOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetectionPEmerged.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_createExtendWindowsWithAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_mapOneRead.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_mapOneReadSpliceGraph.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_mappedFilter.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_maxMappableLength2strands.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_multMapSelect.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_oneRead.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputAlignments.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputSpliceGraphSAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputTranscriptCIGARp.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputTranscriptSAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputTranscriptSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputVariation.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_peOverlapMergeMap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_quantTranscriptome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_stitchPieces.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_stitchWindowSeeds.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_storeAligns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_transformGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_waspMap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAnnotations.h  \n",
            "  inflating: STAR-2.7.10a/source/STAR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SequenceFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SequenceFuns.h  \n",
            "  inflating: STAR-2.7.10a/source/SharedMemory.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SharedMemory.h  \n",
            "   creating: STAR-2.7.10a/source/SimpleGoodTuring/\n",
            "  inflating: STAR-2.7.10a/source/SimpleGoodTuring/ReadMe.txt  \n",
            "  inflating: STAR-2.7.10a/source/SimpleGoodTuring/sgt.h  \n",
            "  inflating: STAR-2.7.10a/source/SjdbClass.h  \n",
            "  inflating: STAR-2.7.10a/source/Solo.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Solo.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloBarcode.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloBarcode.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloBarcode_extractBarcode.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloCommon.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeatureTypes.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_addBAMtags.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_cellFiltering.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_collapseUMI_Graph.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_collapseUMIall_multiMappers.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_countCBgeneUMI.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_countSmartSeq.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_countVelocyto.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_emptyDrops_CR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_loadRawMatrix.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_outputResults.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_processRecords.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_quantTranscript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_redistributeReadsByCB.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_statsOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_sumThreads.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFilteredCells.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloRead.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloRead.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcode.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcode.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcodeStats.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcode_getCBandUMI.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeatureStats.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature_inputRecords.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature_record.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloRead_record.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph.h  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph_findSuperTr.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph_swScoreSpliced.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph_swTraceBack.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Stats.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Stats.h  \n",
            "  inflating: STAR-2.7.10a/source/SuffixArrayFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SuffixArrayFuns.h  \n",
            "  inflating: STAR-2.7.10a/source/SuperTranscriptome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SuperTranscriptome.h  \n",
            "  inflating: STAR-2.7.10a/source/Test.hpp  \n",
            "  inflating: STAR-2.7.10a/source/ThreadControl.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ThreadControl.h  \n",
            "  inflating: STAR-2.7.10a/source/TimeFunctions.cpp  \n",
            "  inflating: STAR-2.7.10a/source/TimeFunctions.h  \n",
            "  inflating: STAR-2.7.10a/source/Transcript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript.h  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_alignScore.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_convertGenomeCigar.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_generateCigarP.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_transformGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_variationAdjust.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_variationOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome.h  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_alignExonOverlap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_classifyAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_geneCountsAddAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_geneFullAlignOverlap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_quantAlign.cpp  \n",
            " extracting: STAR-2.7.10a/source/VERSION  \n",
            "  inflating: STAR-2.7.10a/source/Variation.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Variation.h  \n",
            "  inflating: STAR-2.7.10a/source/bamRemoveDuplicates.cpp  \n",
            "  inflating: STAR-2.7.10a/source/bamRemoveDuplicates.h  \n",
            "  inflating: STAR-2.7.10a/source/bamSortByCoordinate.cpp  \n",
            "  inflating: STAR-2.7.10a/source/bamSortByCoordinate.h  \n",
            "  inflating: STAR-2.7.10a/source/bam_cat.c  \n",
            "  inflating: STAR-2.7.10a/source/bam_cat.h  \n",
            "  inflating: STAR-2.7.10a/source/binarySearch2.cpp  \n",
            "  inflating: STAR-2.7.10a/source/binarySearch2.h  \n",
            "  inflating: STAR-2.7.10a/source/blocksOverlap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/blocksOverlap.h  \n",
            "  inflating: STAR-2.7.10a/source/extendAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/extendAlign.h  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixes.cpp  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixes.h  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixesMemcmp.cpp  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixesMemcmp.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeGenerate.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeParametersWrite.cpp  \n",
            "  inflating: STAR-2.7.10a/source/genomeParametersWrite.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeSAindex.cpp  \n",
            "  inflating: STAR-2.7.10a/source/genomeSAindex.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeScanFastaFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/genomeScanFastaFiles.h  \n",
            "   creating: STAR-2.7.10a/source/htslib/\n",
            "  inflating: STAR-2.7.10a/source/htslib/.gitignore  \n",
            "  inflating: STAR-2.7.10a/source/htslib/.travis.yml  \n",
            "  inflating: STAR-2.7.10a/source/htslib/Makefile  \n",
            "  inflating: STAR-2.7.10a/source/htslib/README.md  \n",
            "  inflating: STAR-2.7.10a/source/htslib/bgzf.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/bgzip.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/config.h  \n",
            "   creating: STAR-2.7.10a/source/htslib/cram/\n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_codecs.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_codecs.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_decode.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_decode.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_encode.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_encode.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_index.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_index.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_io.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_io.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_samtools.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_samtools.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_stats.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_stats.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_structs.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/files.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/mFILE.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/mFILE.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/md5.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/md5.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/misc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/open_trace_file.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/open_trace_file.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/os.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/pooled_alloc.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/pooled_alloc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/sam_header.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/sam_header.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/string_alloc.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/string_alloc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/thread_pool.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/thread_pool.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/vlen.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/vlen.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/zfio.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/zfio.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/faidx.5  \n",
            "  inflating: STAR-2.7.10a/source/htslib/faidx.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hfile.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hfile_internal.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hfile_net.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hts.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib.mk  \n",
            "   creating: STAR-2.7.10a/source/htslib/htslib/\n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/bgzf.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/faidx.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/hfile.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/hts.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/hts_defs.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kfunc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/khash.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/khash_str2int.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/klist.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/knetfile.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kseq.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/ksort.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kstdint.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kstring.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/sam.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/synced_bcf_reader.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/tbx.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/vcf.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/vcf_sweep.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/vcfutils.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib_vars.mk  \n",
            "  inflating: STAR-2.7.10a/source/htslib/kfunc.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/knetfile.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/kstring.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/sam.5  \n",
            "  inflating: STAR-2.7.10a/source/htslib/sam.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/synced_bcf_reader.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/tabix.1  \n",
            "  inflating: STAR-2.7.10a/source/htslib/tabix.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/tbx.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcf.5  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcf.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcf_sweep.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcfutils.c  \n",
            "  inflating: STAR-2.7.10a/source/insertSeqSA.cpp  \n",
            "  inflating: STAR-2.7.10a/source/insertSeqSA.h  \n",
            "  inflating: STAR-2.7.10a/source/mapThreadsSpawn.cpp  \n",
            "  inflating: STAR-2.7.10a/source/mapThreadsSpawn.h  \n",
            "   creating: STAR-2.7.10a/source/opal/\n",
            "  inflating: STAR-2.7.10a/source/opal/LICENSE  \n",
            "  inflating: STAR-2.7.10a/source/opal/opal.cpp  \n",
            "  inflating: STAR-2.7.10a/source/opal/opal.h  \n",
            "  inflating: STAR-2.7.10a/source/opal/simde_avx2.h  \n",
            "  inflating: STAR-2.7.10a/source/outputSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/outputSJ.h  \n",
            "  inflating: STAR-2.7.10a/source/parametersDefault  \n",
            "  inflating: STAR-2.7.10a/source/parametersDefault.xxd  \n",
            "  inflating: STAR-2.7.10a/source/readBarcodeLoad.h  \n",
            "  inflating: STAR-2.7.10a/source/readLoad.cpp  \n",
            "  inflating: STAR-2.7.10a/source/readLoad.h  \n",
            "  inflating: STAR-2.7.10a/source/samHeaders.cpp  \n",
            "  inflating: STAR-2.7.10a/source/samHeaders.h  \n",
            "  inflating: STAR-2.7.10a/source/serviceFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/signalFromBAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/signalFromBAM.h  \n",
            "  inflating: STAR-2.7.10a/source/sjAlignSplit.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjAlignSplit.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbBuildIndex.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbBuildIndex.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbInsertJunctions.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbInsertJunctions.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromFiles.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromStream.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromStream.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbPrepare.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbPrepare.h  \n",
            "  inflating: STAR-2.7.10a/source/soloInputFeatureUMI.cpp  \n",
            "  inflating: STAR-2.7.10a/source/soloInputFeatureUMI.h  \n",
            "  inflating: STAR-2.7.10a/source/sortSuffixesBucket.h  \n",
            "  inflating: STAR-2.7.10a/source/stitchAlignToTranscript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stitchAlignToTranscript.h  \n",
            "  inflating: STAR-2.7.10a/source/stitchGapIndel.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stitchWindowAligns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stitchWindowAligns.h  \n",
            "  inflating: STAR-2.7.10a/source/streamFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/streamFuns.h  \n",
            "  inflating: STAR-2.7.10a/source/stringSubstituteAll.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stringSubstituteAll.h  \n",
            "  inflating: STAR-2.7.10a/source/sysRemoveDir.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sysRemoveDir.h  \n",
            "  inflating: STAR-2.7.10a/source/twoPassRunPass1.cpp  \n",
            "  inflating: STAR-2.7.10a/source/twoPassRunPass1.h  \n",
            "make -C htslib lib-static\n",
            "make[1]: Entering directory '/content/STAR-2.7.10a/source/htslib'\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o kfunc.o kfunc.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o knetfile.o knetfile.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o kstring.o kstring.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o bgzf.o bgzf.c\n",
            "\u001b[01m\u001b[Kbgzf.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kbgzf_write\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kbgzf.c:733:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  733 |     \u001b[01;35m\u001b[Kif\u001b[m\u001b[K ( !fp->is_compressed )\n",
            "      |     \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kbgzf.c:736:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n",
            "  736 |         \u001b[01;36m\u001b[Kconst\u001b[m\u001b[K uint8_t *input = (const uint8_t*)data;\n",
            "      |         \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o faidx.o faidx.c\n",
            "\u001b[01m\u001b[Kfaidx.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kfai_load\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kfaidx.c:265:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kelse\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  265 |     \u001b[01;35m\u001b[Kelse\u001b[m\u001b[K\n",
            "      |     \u001b[01;35m\u001b[K^~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kfaidx.c:268:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kelse\u001b[m\u001b[K’\n",
            "  268 |         \u001b[01;36m\u001b[Kif\u001b[m\u001b[K (fp == 0) {\n",
            "      |         \u001b[01;36m\u001b[K^~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hfile.o hfile.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hfile_net.o hfile_net.c\n",
            "echo '#define HTS_VERSION \"0.0.1\"' > version.h\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hts.o hts.c\n",
            "\u001b[01m\u001b[Khts.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Khts_idx_init\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Khts.c:522:63:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Koverflow in conversion from ‘\u001b[01m\u001b[Kuint32_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kunsigned int\u001b[m\u001b[K’} to ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ changes value from ‘\u001b[01m\u001b[Kidx->z.last_bin = 4294967295\u001b[m\u001b[K’ to ‘\u001b[01m\u001b[K-1\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverflow\u0007-Woverflow\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  522 |         idx->z.save_bin = idx->z.save_tid = idx->z.last_tid = \u001b[01;35m\u001b[Kidx\u001b[m\u001b[K->z.last_bin = 0xffffffffu;\n",
            "      |                                                               \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o sam.o sam.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksam.c:3\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kbam_hdr_write\u001b[m\u001b[K’ at \u001b[01m\u001b[Ksam.c:130:2\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying 4 bytes from a string of the same length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcf_sweep.o vcf_sweep.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o tbx.o tbx.c\n",
            "\u001b[01m\u001b[Ktbx.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ktbx_seqnames\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ktbx.c:285:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kfor\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  285 |     \u001b[01;35m\u001b[Kfor\u001b[m\u001b[K (tid=0; tid<m; tid++)\n",
            "      |     \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ktbx.c:287:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kfor\u001b[m\u001b[K’\n",
            "  287 |         \u001b[01;36m\u001b[K*\u001b[m\u001b[Kn = m;\n",
            "      |         \u001b[01;36m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcf.o vcf.c\n",
            "\u001b[01m\u001b[Kvcf.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kvcf_format\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kvcf.c:1909:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1909 |                         \u001b[01;35m\u001b[Kif\u001b[m\u001b[K ( !first ) kputc(';', s); first = 0;\n",
            "      |                         \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kvcf.c:1909:54:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n",
            " 1909 |                         if ( !first ) kputc(';', s); \u001b[01;36m\u001b[Kfirst\u001b[m\u001b[K = 0;\n",
            "      |                                                      \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kvcf.c:1948:21:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1948 |                     \u001b[01;35m\u001b[Kif\u001b[m\u001b[K (!first) kputc(':', s); first = 0;\n",
            "      |                     \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kvcf.c:1948:48:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n",
            " 1948 |                     if (!first) kputc(':', s); \u001b[01;36m\u001b[Kfirst\u001b[m\u001b[K = 0;\n",
            "      |                                                \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcfutils.o vcfutils.c\n",
            "In file included from \u001b[01m\u001b[Khtslib/vcf.h:12\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Khtslib/vcfutils.h:8\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kvcfutils.c:1\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Kvcfutils.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kbcf_remove_alleles\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kvcfutils.c:291:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  291 |                 assert( \u001b[01;35m\u001b[Kn\u001b[m\u001b[K=nG_ori );\n",
            "      |                         \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_index.o cram/cram_index.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_io.o cram/cram_io.c\n",
            "\u001b[01m\u001b[Kcram/cram_io.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/cram_io.c:1529:34:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K.tmp_\u001b[m\u001b[K’ directive writing 5 bytes into a region of size between 1 and 4096 [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wformat-overflow=\u0007-Wformat-overflow=\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1529 |             sprintf(path_tmp, \"%s\u001b[01;35m\u001b[K.tmp_\u001b[m\u001b[K%d\", path, /*getpid(),*/ i);\n",
            "      |                                  \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/cram_io.c:1529:31:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdirective argument in the range [0, 2147483647]\n",
            " 1529 |             sprintf(path_tmp, \u001b[01;36m\u001b[K\"%s.tmp_%d\"\u001b[m\u001b[K, path, /*getpid(),*/ i);\n",
            "      |                               \u001b[01;36m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/stdio.h:894\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:51\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 7 and 4111 bytes into a destination of size 4096\n",
            "   38 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n",
            "      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   39 | \u001b[01;36m\u001b[K                                  __glibc_objsize (__s), __fmt,\u001b[m\u001b[K\n",
            "      |                                   \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   40 | \u001b[01;36m\u001b[K                                  __va_arg_pack ())\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:55\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2917:6\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2987:3\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2909:2\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2987:3\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_dopen\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3291:2\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 20 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_samtools.o cram/cram_samtools.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/files.o cram/files.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/mFILE.o cram/mFILE.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/md5.o cram/md5.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/sam_header.o cram/sam_header.c\n",
            "In file included from \u001b[01m\u001b[Kcram/sam_header.c:36\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ksam_hdr_add_lines\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:328:20:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  328 |             assert(\u001b[01;35m\u001b[Kp\u001b[m\u001b[K->next = t);\n",
            "      |                    \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ksam_hdr_vadd\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:449:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  449 |         assert(\u001b[01;35m\u001b[Kp\u001b[m\u001b[K->next = t);\n",
            "      |                \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/string_alloc.c:39\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_ndup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:149:5\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:141:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying as many bytes from a string as its length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:141:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Klength computed here\n",
            "  141 |     return \u001b[01;36m\u001b[Kstring_ndup(a_str, instr, strlen(instr))\u001b[m\u001b[K;\n",
            "      |            \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/thread_pool.o cram/thread_pool.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/vlen.o cram/vlen.c\n",
            "\u001b[01m\u001b[Kcram/vlen.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kvflen\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/vlen.c:121:9:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Ki\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  121 |     int \u001b[01;35m\u001b[Ki\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/zfio.o cram/zfio.c\n",
            "ar -rc libhts.a kfunc.o knetfile.o kstring.o bgzf.o faidx.o hfile.o hfile_net.o hts.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/md5.o cram/open_trace_file.o cram/pooled_alloc.o cram/sam_header.o cram/string_alloc.o cram/thread_pool.o cram/vlen.o cram/zfio.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/STAR-2.7.10a/source/htslib'\n",
            "echo BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall_multiMappers.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c\n",
            "BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall_multiMappers.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c\n",
            "'rm' -f ./Depend.list\n",
            "g++ -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:30+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -MM BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall_multiMappers.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c parametersDefault.xxd htslib >> Depend.list\n",
            "\u001b[01m\u001b[Kg++:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[KparametersDefault.xxd: linker input file unused because linking not done\n",
            "\u001b[01m\u001b[Kg++:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Khtslib: linker input file unused because linking not done\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_collapseUMI_Graph.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_collapseUMIall_multiMappers.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersClip_initialize.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipMate_clip.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipCR4.cpp\n",
            "cd opal && \\\n",
            "g++ -c -I./ -std=c++11   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra  -mavx2 opal.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipMate_clipChunk.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipMate_initialize.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_loadRawMatrix.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_emptyDrops_CR.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra soloInputFeatureUMI.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_countSmartSeq.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_redistributeReadsByCB.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_quantTranscript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_sumThreads.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_countVelocyto.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_countCBgeneUMI.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_classifyAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_cellFiltering.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_statsOutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra bamSortByCoordinate.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloBarcode.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersSolo.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloRead.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloRead_record.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadBarcode.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadBarcode_getCBandUMI.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloBarcode_extractBarcode.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadFeature.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadFeature_record.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadFeature_inputRecords.cpp\n",
            "\u001b[01m\u001b[KSoloReadFeature_inputRecords.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid SoloReadFeature::inputRecords(unsigned int**, unsigned int, std::vector<unsigned int>&, std::vector<readInfoStruct>&, SoloReadFlagClass&, std::vector<unsigned int>&, std::vector<unsigned int>&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSoloReadFeature_inputRecords.cpp:21:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kunused variable ‘\u001b[01m\u001b[KnReadsIn\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable\u0007-Wunused-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   21 |     uint64 \u001b[01;35m\u001b[KnReadsIn\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~\u001b[m\u001b[K\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Solo.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_outputResults.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_processRecords.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_addBAMtags.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_transformGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_transformGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_convertGenomeCigar.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra twoPassRunPass1.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra samHeaders.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_genomeLoad.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_genomeOutLoad.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_transformGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputSpliceGraphSAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_mapOneReadSpliceGraph.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph_swScoreSpliced.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph_swTraceBack.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph_findSuperTr.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjAlignSplit.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra GTF.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra GTF_transcriptGeneSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra GTF_superTranscript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SuperTranscriptome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputAlignments.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra STAR.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SharedMemory.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra PackedArray.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SuffixArrayFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters_samAttributes.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra InOutStreams.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SequenceFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Stats.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_alignScore.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_generateCigarP.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Chain.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_variationAdjust.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Variation.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_waspMap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_storeAligns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_stitchPieces.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_multMapSelect.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_mapOneRead.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra readLoad.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlignChunk.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlignChunk_processChunks.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlignChunk_mapChunk.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra OutSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra outputSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra blocksOverlap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ThreadControl.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sysRemoveDir.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_maxMappableLength2strands.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra binarySearch2.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputTranscriptSAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputTranscriptSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputTranscriptCIGARp.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_createExtendWindowsWithAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_assignAlignToWindow.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_oneRead.cpp\n",
            "\u001b[01m\u001b[KReadAlign_oneRead.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kint ReadAlign::oneRead()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KReadAlign_oneRead.cpp:23:21:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K*readStatus[0]\u001b[m\u001b[K’ may be used uninitialized [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmaybe-uninitialized\u0007-Wmaybe-uninitialized\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   23 |     if (\u001b[01;35m\u001b[KreadStatus[0]\u001b[m\u001b[K==-1) {//finished with the stream\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~~~~~^\u001b[m\u001b[K\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_stitchWindowSeeds.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_peOverlapMergeMap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_mappedFilter.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersChimeric_initialize.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetection.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetectionOld.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetectionOldOutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericDetection.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericDetection_chimericDetectionMult.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetectionPEmerged.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra stitchWindowAligns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra extendAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra stitchAlignToTranscript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericAlign_chimericJunctionOutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericAlign_chimericBAMoutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericAlign_chimericStitching.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_genomeGenerate.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra genomeParametersWrite.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra genomeScanFastaFiles.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra genomeSAindex.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_insertSequences.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra insertSeqSA.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra funCompareUintAndSuffixes.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra funCompareUintAndSuffixesMemcmp.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra TimeFunctions.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ErrorWarning.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra streamFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra stringSubstituteAll.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_quantAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_geneFullAlignOverlap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_quantTranscriptome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Quantifications.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_geneCountsAddAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbLoadFromFiles.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbLoadFromStream.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbPrepare.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbBuildIndex.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbInsertJunctions.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra mapThreadsSpawn.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters_readFilesInit.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters_readSAMheader.cpp\n",
            "\u001b[01m\u001b[KParameters_readSAMheader.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid Parameters::readSAMheader(std::string, std::vector<std::__cxx11::basic_string<char> >)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KParameters_readSAMheader.cpp:32:15:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint system(const char*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |         \u001b[01;35m\u001b[Ksystem(com1.c_str())\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "g++ -c  -pipe -Wall -Wextra -O3 bam_cat.c\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra serviceFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMoutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMfunctions.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_alignBAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMbinSortByCoordinate.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra signalFromBAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra bamRemoveDuplicates.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMbinSortUnmapped.cpp\n",
            "g++ -o STAR  -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T05:09:48+00:00 f28f9d2ec208:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_collapseUMI_Graph.o SoloFeature_collapseUMIall_multiMappers.o ParametersClip_initialize.o ClipMate_clip.o ClipCR4.o opal/opal.o ClipMate_clipChunk.o ClipMate_initialize.o SoloFeature_loadRawMatrix.o SoloFeature_emptyDrops_CR.o soloInputFeatureUMI.o SoloFeature_countSmartSeq.o SoloFeature_redistributeReadsByCB.o SoloFeature_quantTranscript.o SoloFeature_sumThreads.o SoloFeature_countVelocyto.o SoloFeature_countCBgeneUMI.o Transcriptome_classifyAlign.o Transcriptome_geneFullAlignOverlap_ExonOverIntron.o Transcriptome_alignExonOverlap.cpp SoloFeature_cellFiltering.o SoloFeature_statsOutput.o bamSortByCoordinate.o SoloBarcode.o ParametersSolo.o SoloRead.o SoloRead_record.o SoloReadBarcode.o SoloReadBarcode_getCBandUMI.o SoloBarcode_extractBarcode.o SoloReadFeature.o SoloReadFeature_record.o SoloReadFeature_inputRecords.o Solo.o SoloFeature.o SoloFeature_outputResults.o SoloFeature_processRecords.o SoloFeature_addBAMtags.o ReadAlign_transformGenome.o Genome_transformGenome.o Transcript_convertGenomeCigar.o twoPassRunPass1.o samHeaders.o Genome_genomeLoad.o Genome_genomeOutLoad.o Transcript_transformGenome.o ReadAlign_outputSpliceGraphSAM.o ReadAlign_mapOneReadSpliceGraph.o SpliceGraph.o SpliceGraph_swScoreSpliced.o SpliceGraph_swTraceBack.o SpliceGraph_findSuperTr.o sjAlignSplit.o GTF.o GTF_transcriptGeneSJ.o GTF_superTranscript.o SuperTranscriptome.o ReadAlign_outputAlignments.o ReadAlign.o STAR.o SharedMemory.o PackedArray.o SuffixArrayFuns.o Parameters.o Parameters_samAttributes.o InOutStreams.o SequenceFuns.o Genome.o ParametersGenome.o Stats.o Transcript.o Transcript_alignScore.o Transcript_generateCigarP.o Chain.o Transcript_variationAdjust.o Variation.o ReadAlign_waspMap.o ReadAlign_storeAligns.o ReadAlign_stitchPieces.o ReadAlign_multMapSelect.o ReadAlign_mapOneRead.o readLoad.o ReadAlignChunk.o ReadAlignChunk_processChunks.o ReadAlignChunk_mapChunk.o OutSJ.o outputSJ.o blocksOverlap.o ThreadControl.o sysRemoveDir.o ReadAlign_maxMappableLength2strands.o binarySearch2.o ReadAlign_outputTranscriptSAM.o ReadAlign_outputTranscriptSJ.o ReadAlign_outputTranscriptCIGARp.o ReadAlign_calcCIGAR.cpp ReadAlign_createExtendWindowsWithAlign.o ReadAlign_assignAlignToWindow.o ReadAlign_oneRead.o ReadAlign_stitchWindowSeeds.o ReadAlign_peOverlapMergeMap.o ReadAlign_mappedFilter.o ParametersChimeric_initialize.o ReadAlign_chimericDetection.o ReadAlign_chimericDetectionOld.o ReadAlign_chimericDetectionOldOutput.o ChimericDetection.o ChimericDetection_chimericDetectionMult.o ReadAlign_chimericDetectionPEmerged.o stitchWindowAligns.o extendAlign.o stitchAlignToTranscript.o ChimericSegment.cpp ChimericAlign.cpp ChimericAlign_chimericJunctionOutput.o ChimericAlign_chimericBAMoutput.o ChimericAlign_chimericStitching.o Genome_genomeGenerate.o genomeParametersWrite.o genomeScanFastaFiles.o genomeSAindex.o Genome_insertSequences.o insertSeqSA.o funCompareUintAndSuffixes.o funCompareUintAndSuffixesMemcmp.o TimeFunctions.o ErrorWarning.o streamFuns.o stringSubstituteAll.o Transcriptome.o Transcriptome_quantAlign.o Transcriptome_geneFullAlignOverlap.o ReadAlign_quantTranscriptome.o Quantifications.o Transcriptome_geneCountsAddAlign.o sjdbLoadFromFiles.o sjdbLoadFromStream.o sjdbPrepare.o sjdbBuildIndex.o sjdbInsertJunctions.o mapThreadsSpawn.o Parameters_readFilesInit.o Parameters_openReadsFiles.cpp Parameters_closeReadsFiles.cpp Parameters_readSAMheader.o bam_cat.o serviceFuns.o GlobalVariables.cpp BAMoutput.o BAMfunctions.o ReadAlign_alignBAM.o BAMbinSortByCoordinate.o signalFromBAM.o bamRemoveDuplicates.o BAMbinSortUnmapped.o  -pthread -Lhtslib -Bstatic -lhts -Bdynamic -lz \n",
            "\u001b[01m\u001b[KParameters_openReadsFiles.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid Parameters::openReadsFiles()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KParameters_openReadsFiles.cpp:25:20:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kimate\u001b[m\u001b[K’ might be clobbered by ‘\u001b[01m\u001b[Klongjmp\u001b[m\u001b[K’ or ‘\u001b[01m\u001b[Kvfork\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wclobbered\u0007-Wclobbered\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   25 |          for (uint \u001b[01;35m\u001b[Kimate\u001b[m\u001b[K=0;imate<readFilesNames.size();imate++) {//open readIn files\n",
            "      |                    \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n"
          ]
        }
      ],
      "source": [
        "# Step 0: Install Required Tools and Libraries\n",
        "# Update package list\n",
        "!apt-get update\n",
        "\n",
        "# Install build tools and SRA Toolkit\n",
        "!apt-get install -y build-essential sra-toolkit unzip\n",
        "\n",
        "# Download and compile STAR\n",
        "!wget https://github.com/alexdobin/STAR/archive/2.7.10a.zip\n",
        "!unzip 2.7.10a.zip\n",
        "!cd STAR-2.7.10a/source && make STAR\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Step 1: Download FASTQ Files from SRA\n",
        "# Download and convert SRR11548957 to paired-end FASTQ files\n",
        "!prefetch SRR11548957\n",
        "!fasterq-dump SRR11548957 --split-files"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RYnHsucqUm8b",
        "outputId": "bab20748-2556-464e-a055-fe1ab90c47d0"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "2025-11-11T05:14:13 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2025-11-11T05:14:14 prefetch.2.11.3: 1) 'SRR11548957' is found locally\n",
            "2025-11-11T05:14:14 prefetch.2.11.3: 'SRR11548957' has 0 unresolved dependencies\n",
            "spots read      : 1,792,464\n",
            "reads read      : 3,584,928\n",
            "reads written   : 3,584,928\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Step 2: Prepare Reference Genome (chr22)\n",
        "# Create reference directory and download chr22 from Ensembl\n",
        "!mkdir -p reference\n",
        "!wget -P reference ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n",
        "!wget -P reference ftp://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz\n",
        "\n",
        "# Unzip the FASTA and GTF files\n",
        "!gunzip reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n",
        "!gunzip reference/Homo_sapiens.GRCh38.109.gtf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tNjS-Bf-UqPO",
        "outputId": "65e2b546-03b3-440a-f83b-24ed5f08a419"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-11-11 05:16:00--  ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n",
            "           => ‘reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz’\n",
            "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
            "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /pub/release-109/fasta/homo_sapiens/dna ... done.\n",
            "==> SIZE Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz ... 11389810\n",
            "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz ... done.\n",
            "Length: 11389810 (11M) (unauthoritative)\n",
            "\n",
            "Homo_sapiens.GRCh38 100%[===================>]  10.86M  9.30MB/s    in 1.2s    \n",
            "\n",
            "2025-11-11 05:16:03 (9.30 MB/s) - ‘reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz’ saved [11389810]\n",
            "\n",
            "--2025-11-11 05:16:03--  ftp://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz\n",
            "           => ‘reference/Homo_sapiens.GRCh38.109.gtf.gz’\n",
            "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
            "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /pub/release-109/gtf/homo_sapiens ... done.\n",
            "==> SIZE Homo_sapiens.GRCh38.109.gtf.gz ... 54258835\n",
            "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.109.gtf.gz ... done.\n",
            "Length: 54258835 (52M) (unauthoritative)\n",
            "\n",
            "Homo_sapiens.GRCh38 100%[===================>]  51.75M  20.1MB/s    in 2.6s    \n",
            "\n",
            "2025-11-11 05:16:07 (20.1 MB/s) - ‘reference/Homo_sapiens.GRCh38.109.gtf.gz’ saved [54258835]\n",
            "\n",
            "gzip: reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa already exists; do you wish to overwrite (y or n)? n\n",
            "\tnot overwritten\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#  Step 3: Generate STAR Genome Index\n",
        "# Create index directory and build genome index\n",
        "!mkdir -p star_index\n",
        "!/content/STAR-2.7.10a/bin/Linux_x86_64_static/STAR \\\n",
        "    --runThreadN 4 \\\n",
        "    --runMode genomeGenerate \\\n",
        "    --genomeDir star_index \\\n",
        "    --genomeFastaFiles reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa \\\n",
        "    --sjdbGTFfile reference/Homo_sapiens.GRCh38.109.gtf \\\n",
        "    --genomeSAindexNbases 9"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gBnhnC0OUtDk",
        "outputId": "53823e74-3ea7-43fa-f46f-e281664e2cde"
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\t/content/STAR-2.7.10a/bin/Linux_x86_64_static/STAR --runThreadN 4 --runMode genomeGenerate --genomeDir star_index --genomeFastaFiles reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa --sjdbGTFfile reference/Homo_sapiens.GRCh38.109.gtf --genomeSAindexNbases 9\n",
            "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
            "Nov 11 05:16:36 ..... started STAR run\n",
            "Nov 11 05:16:36 ... starting to generate Genome files\n",
            "Nov 11 05:16:37 ..... processing annotations GTF\n",
            "Nov 11 05:16:44 ... starting to sort Suffix Array. This may take a long time...\n",
            "Nov 11 05:16:44 ... sorting Suffix Array chunks and saving them to disk...\n",
            "Nov 11 05:16:53 ... loading chunks from disk, packing SA...\n",
            "Nov 11 05:16:54 ... finished generating suffix array\n",
            "Nov 11 05:16:54 ... generating Suffix Array index\n",
            "Nov 11 05:16:54 ... completed Suffix Array index\n",
            "Nov 11 05:16:54 ..... inserting junctions into the genome indices\n",
            "Nov 11 05:16:58 ... writing Genome to disk ...\n",
            "Nov 11 05:17:00 ... writing Suffix Array to disk ...\n",
            "Nov 11 05:17:00 ... writing SAindex to disk\n",
            "Nov 11 05:17:00 ..... finished successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#  Step 4: Run STAR Alignment\n",
        "# Create output directory and align reads to reference genome\n",
        "!mkdir -p star_output\n",
        "!/content/STAR-2.7.10a/bin/Linux_x86_64_static/STAR \\\n",
        "    --runThreadN 4 \\\n",
        "    --genomeDir star_index \\\n",
        "    --readFilesIn SRR11548957_1.fastq SRR11548957_2.fastq \\\n",
        "    --outFileNamePrefix star_output/SRR11548957_ \\\n",
        "    --outSAMtype BAM SortedByCoordinate \\\n",
        "    --quantMode GeneCounts"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DZunGK1hUwMd",
        "outputId": "d705f43d-84ba-4b3a-abd7-8163dc827076"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\t/content/STAR-2.7.10a/bin/Linux_x86_64_static/STAR --runThreadN 4 --genomeDir star_index --readFilesIn SRR11548957_1.fastq SRR11548957_2.fastq --outFileNamePrefix star_output/SRR11548957_ --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts\n",
            "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
            "Nov 11 05:17:18 ..... started STAR run\n",
            "Nov 11 05:17:18 ..... loading genome\n",
            "Nov 11 05:17:18 ..... started mapping\n",
            "Nov 11 05:39:57 ..... finished mapping\n",
            "Nov 11 05:39:58 ..... started sorting BAM\n",
            "Nov 11 05:39:58 ..... finished successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#  Step 5: Inspect Output Files\n",
        "# List key output files\n",
        "!ls -lh star_output/SRR11548957_Aligned.sortedByCoord.out.bam\n",
        "!ls -lh star_output/SRR11548957_ReadsPerGene.out.tab\n",
        "!cat star_output/SRR11548957_Log.final.out\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HgFQpfTtUyuV",
        "outputId": "c80d0e17-4714-4826-ca6a-cf192991168e"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "-rw-r--r-- 1 root root 19M Nov 11 05:39 star_output/SRR11548957_Aligned.sortedByCoord.out.bam\n",
            "-rw-r--r-- 1 root root 32K Nov 11 05:39 star_output/SRR11548957_ReadsPerGene.out.tab\n",
            "                                 Started job on |\tNov 11 05:17:18\n",
            "                             Started mapping on |\tNov 11 05:17:18\n",
            "                                    Finished on |\tNov 11 05:39:58\n",
            "       Mapping speed, Million of reads per hour |\t4.74\n",
            "\n",
            "                          Number of input reads |\t1792464\n",
            "                      Average input read length |\t150\n",
            "                                    UNIQUE READS:\n",
            "                   Uniquely mapped reads number |\t156947\n",
            "                        Uniquely mapped reads % |\t8.76%\n",
            "                          Average mapped length |\t141.22\n",
            "                       Number of splices: Total |\t12073\n",
            "            Number of splices: Annotated (sjdb) |\t6261\n",
            "                       Number of splices: GT/AG |\t10520\n",
            "                       Number of splices: GC/AG |\t209\n",
            "                       Number of splices: AT/AC |\t7\n",
            "               Number of splices: Non-canonical |\t1337\n",
            "                      Mismatch rate per base, % |\t5.16%\n",
            "                         Deletion rate per base |\t0.11%\n",
            "                        Deletion average length |\t2.96\n",
            "                        Insertion rate per base |\t0.17%\n",
            "                       Insertion average length |\t2.05\n",
            "                             MULTI-MAPPING READS:\n",
            "        Number of reads mapped to multiple loci |\t29782\n",
            "             % of reads mapped to multiple loci |\t1.66%\n",
            "        Number of reads mapped to too many loci |\t1463\n",
            "             % of reads mapped to too many loci |\t0.08%\n",
            "                                  UNMAPPED READS:\n",
            "  Number of reads unmapped: too many mismatches |\t0\n",
            "       % of reads unmapped: too many mismatches |\t0.00%\n",
            "            Number of reads unmapped: too short |\t1602192\n",
            "                 % of reads unmapped: too short |\t89.38%\n",
            "                Number of reads unmapped: other |\t2080\n",
            "                     % of reads unmapped: other |\t0.12%\n",
            "                                  CHIMERIC READS:\n",
            "                       Number of chimeric reads |\t0\n",
            "                            % of chimeric reads |\t0.00%\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Step 6: Download Output Files (Colab)\n",
        "from google.colab import files\n",
        "files.download('star_output/SRR11548957_Aligned.sortedByCoord.out.bam')\n",
        "files.download('star_output/SRR11548957_ReadsPerGene.out.tab')\n",
        "files.download('star_output/SRR11548957_Log.final.out')\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 17
        },
        "id": "WYVu_CtiU1eS",
        "outputId": "16a0c51f-f34d-4a2d-a87a-da453cd09ee4"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_7f1d94f9-b315-486a-880b-f008997f4521\", \"SRR11548957_Aligned.sortedByCoord.out.bam\", 19682923)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_1f0d43e0-2d7e-4afd-a280-500ee5844507\", \"SRR11548957_ReadsPerGene.out.tab\", 32456)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_8eadd8cf-3af1-42b7-aec5-2c9da6b8f607\", \"SRR11548957_Log.final.out\", 2007)"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3a3c3d25"
      },
      "source": [
        "# Task\n",
        "Generate a Plotly plot of the STAR alignment results from the log file \"star_output/SRR11548957_Log.final.out\"."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8630c44c"
      },
      "source": [
        "## Parse log file\n",
        "\n",
        "### Subtask:\n",
        "Read the `star_output/SRR11548957_Log.final.out` file and extract the relevant mapping statistics (e.g., unique mapping percentage, multi-mapping percentage, unmapped percentages).\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6450f4be"
      },
      "source": [
        "**Reasoning**:\n",
        "I need to read the log file and extract the relevant mapping statistics to prepare for plotting. I will use Python to open and parse the file, extracting the required percentages.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bcfb79b5",
        "outputId": "8f60ea0e-abea-4511-fd64-0b6218107d06"
      },
      "source": [
        "import pandas as pd\n",
        "\n",
        "log_file_path = 'star_output/SRR11548957_Log.final.out'\n",
        "mapping_stats = {}\n",
        "\n",
        "with open(log_file_path, 'r') as f:\n",
        "    for line in f:\n",
        "        if 'Uniquely mapped reads %' in line:\n",
        "            mapping_stats['Uniquely mapped reads %'] = float(line.split('|')[-1].strip().replace('%', ''))\n",
        "        elif '% of reads mapped to multiple loci' in line:\n",
        "            mapping_stats['% of reads mapped to multiple loci'] = float(line.split('|')[-1].strip().replace('%', ''))\n",
        "        elif '% of reads mapped to too many loci' in line:\n",
        "            mapping_stats['% of reads mapped to too many loci'] = float(line.split('|')[-1].strip().replace('%', ''))\n",
        "        elif '% of reads unmapped: too many mismatches' in line:\n",
        "            mapping_stats['% of reads unmapped: too many mismatches'] = float(line.split('|')[-1].strip().replace('%', ''))\n",
        "        elif '% of reads unmapped: too short' in line:\n",
        "            mapping_stats['% of reads unmapped: too short'] = float(line.split('|')[-1].strip().replace('%', ''))\n",
        "        elif '% of reads unmapped: other' in line:\n",
        "            mapping_stats['% of reads unmapped: other'] = float(line.split('|')[-1].strip().replace('%', ''))\n",
        "\n",
        "print(mapping_stats)"
      ],
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'Uniquely mapped reads %': 8.76, '% of reads mapped to multiple loci': 1.66, '% of reads mapped to too many loci': 0.08, '% of reads unmapped: too many mismatches': 0.0, '% of reads unmapped: too short': 89.38, '% of reads unmapped: other': 0.12}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "2a26cbd0"
      },
      "source": [
        "## Prepare data for plotting\n",
        "\n",
        "### Subtask:\n",
        "Organize the extracted statistics into a format suitable for Plotly. A pandas DataFrame would be ideal.\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "aa88bff5"
      },
      "source": [
        "**Reasoning**:\n",
        "Convert the mapping statistics dictionary into a pandas DataFrame with 'Category' and 'Percentage' columns as required.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 237
        },
        "id": "8af5284f",
        "outputId": "915f2ca3-a916-4544-ad8d-28227851f304"
      },
      "source": [
        "mapping_df = pd.DataFrame(list(mapping_stats.items()), columns=['Category', 'Percentage'])\n",
        "display(mapping_df)"
      ],
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "                                   Category  Percentage\n",
              "0                   Uniquely mapped reads %        8.76\n",
              "1        % of reads mapped to multiple loci        1.66\n",
              "2        % of reads mapped to too many loci        0.08\n",
              "3  % of reads unmapped: too many mismatches        0.00\n",
              "4            % of reads unmapped: too short       89.38\n",
              "5                % of reads unmapped: other        0.12"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-ae9d6dbd-81d3-4ecc-a9d4-1bf9d8e295dd\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Category</th>\n",
              "      <th>Percentage</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Uniquely mapped reads %</td>\n",
              "      <td>8.76</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>% of reads mapped to multiple loci</td>\n",
              "      <td>1.66</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>% of reads mapped to too many loci</td>\n",
              "      <td>0.08</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>% of reads unmapped: too many mismatches</td>\n",
              "      <td>0.00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>% of reads unmapped: too short</td>\n",
              "      <td>89.38</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>% of reads unmapped: other</td>\n",
              "      <td>0.12</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-ae9d6dbd-81d3-4ecc-a9d4-1bf9d8e295dd')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-ae9d6dbd-81d3-4ecc-a9d4-1bf9d8e295dd button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-ae9d6dbd-81d3-4ecc-a9d4-1bf9d8e295dd');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-801be486-36a6-4b60-b65f-2de035f23833\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-801be486-36a6-4b60-b65f-2de035f23833')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-801be486-36a6-4b60-b65f-2de035f23833 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "  <div id=\"id_f58bd4dc-cf39-476c-a0ea-cda40539592e\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('mapping_df')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_f58bd4dc-cf39-476c-a0ea-cda40539592e button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('mapping_df');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "mapping_df",
              "summary": "{\n  \"name\": \"mapping_df\",\n  \"rows\": 6,\n  \"fields\": [\n    {\n      \"column\": \"Category\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 6,\n        \"samples\": [\n          \"Uniquely mapped reads %\",\n          \"% of reads mapped to multiple loci\",\n          \"% of reads unmapped: other\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Percentage\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 35.78164818264617,\n        \"min\": 0.0,\n        \"max\": 89.38,\n        \"num_unique_values\": 6,\n        \"samples\": [\n          8.76,\n          1.66,\n          0.12\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "e91c5b07"
      },
      "source": [
        "## Generate plotly plot\n",
        "\n",
        "### Subtask:\n",
        "Create a bar chart or a pie chart using Plotly to visualize the different mapping categories (uniquely mapped, multi-mapped, unmapped).\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "42b7c3ba"
      },
      "source": [
        "**Reasoning**:\n",
        "Generate a pie chart using Plotly to visualize the mapping statistics stored in the `mapping_df` DataFrame.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 542
        },
        "id": "ed7c00e8",
        "outputId": "8eb354e1-9e85-41a1-fa08-e7e863ab0d2a"
      },
      "source": [
        "import plotly.express as px\n",
        "\n",
        "fig = px.pie(mapping_df, values='Percentage', names='Category', title='STAR Alignment Mapping Statistics')\n",
        "fig.show()"
      ],
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<html>\n",
              "<head><meta charset=\"utf-8\" /></head>\n",
              "<body>\n",
              "    <div>            <script src=\"https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_SVG\"></script><script type=\"text/javascript\">if (window.MathJax && window.MathJax.Hub && window.MathJax.Hub.Config) {window.MathJax.Hub.Config({SVG: {font: \"STIX-Web\"}});}</script>                <script type=\"text/javascript\">window.PlotlyConfig = {MathJaxConfig: 'local'};</script>\n",
              "        <script charset=\"utf-8\" src=\"https://cdn.plot.ly/plotly-2.35.2.min.js\"></script>                <div id=\"69b9930d-8c55-4bcc-8c35-554b9a594fa8\" class=\"plotly-graph-div\" style=\"height:525px; width:100%;\"></div>            <script type=\"text/javascript\">                                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById(\"69b9930d-8c55-4bcc-8c35-554b9a594fa8\")) {                    Plotly.newPlot(                        \"69b9930d-8c55-4bcc-8c35-554b9a594fa8\",                        [{\"domain\":{\"x\":[0.0,1.0],\"y\":[0.0,1.0]},\"hovertemplate\":\"Category=%{label}\\u003cbr\\u003ePercentage=%{value}\\u003cextra\\u003e\\u003c\\u002fextra\\u003e\",\"labels\":[\"Uniquely mapped reads %\",\"% of reads mapped to multiple loci\",\"% of reads mapped to too many loci\",\"% of reads unmapped: too many mismatches\",\"% of reads unmapped: too short\",\"% of reads unmapped: other\"],\"legendgroup\":\"\",\"name\":\"\",\"showlegend\":true,\"values\":[8.76,1.66,0.08,0.0,89.38,0.12],\"type\":\"pie\"}],                        {\"template\":{\"data\":{\"histogram2dcontour\":[{\"type\":\"histogram2dcontour\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"choropleth\":[{\"type\":\"choropleth\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"histogram2d\":[{\"type\":\"histogram2d\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"heatmap\":[{\"type\":\"heatmap\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"heatmapgl\":[{\"type\":\"heatmapgl\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"contourcarpet\":[{\"type\":\"contourcarpet\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"contour\":[{\"type\":\"contour\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"surface\":[{\"type\":\"surface\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"mesh3d\":[{\"type\":\"mesh3d\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"scatter\":[{\"fillpattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2},\"type\":\"scatter\"}],\"parcoords\":[{\"type\":\"parcoords\",\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterpolargl\":[{\"type\":\"scatterpolargl\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"bar\":[{\"error_x\":{\"color\":\"#2a3f5f\"},\"error_y\":{\"color\":\"#2a3f5f\"},\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"bar\"}],\"scattergeo\":[{\"type\":\"scattergeo\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterpolar\":[{\"type\":\"scatterpolar\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"histogram\":[{\"marker\":{\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"histogram\"}],\"scattergl\":[{\"type\":\"scattergl\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatter3d\":[{\"type\":\"scatter3d\",\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scattermapbox\":[{\"type\":\"scattermapbox\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterternary\":[{\"type\":\"scatterternary\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scattercarpet\":[{\"type\":\"scattercarpet\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"carpet\":[{\"aaxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"baxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"type\":\"carpet\"}],\"table\":[{\"cells\":{\"fill\":{\"color\":\"#EBF0F8\"},\"line\":{\"color\":\"white\"}},\"header\":{\"fill\":{\"color\":\"#C8D4E3\"},\"line\":{\"color\":\"white\"}},\"type\":\"table\"}],\"barpolar\":[{\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"barpolar\"}],\"pie\":[{\"automargin\":true,\"type\":\"pie\"}]},\"layout\":{\"autotypenumbers\":\"strict\",\"colorway\":[\"#636efa\",\"#EF553B\",\"#00cc96\",\"#ab63fa\",\"#FFA15A\",\"#19d3f3\",\"#FF6692\",\"#B6E880\",\"#FF97FF\",\"#FECB52\"],\"font\":{\"color\":\"#2a3f5f\"},\"hovermode\":\"closest\",\"hoverlabel\":{\"align\":\"left\"},\"paper_bgcolor\":\"white\",\"plot_bgcolor\":\"#E5ECF6\",\"polar\":{\"bgcolor\":\"#E5ECF6\",\"angularaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"radialaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"ternary\":{\"bgcolor\":\"#E5ECF6\",\"aaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"baxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"caxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"coloraxis\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"colorscale\":{\"sequential\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"sequentialminus\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"diverging\":[[0,\"#8e0152\"],[0.1,\"#c51b7d\"],[0.2,\"#de77ae\"],[0.3,\"#f1b6da\"],[0.4,\"#fde0ef\"],[0.5,\"#f7f7f7\"],[0.6,\"#e6f5d0\"],[0.7,\"#b8e186\"],[0.8,\"#7fbc41\"],[0.9,\"#4d9221\"],[1,\"#276419\"]]},\"xaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"automargin\":true,\"zerolinewidth\":2},\"yaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"automargin\":true,\"zerolinewidth\":2},\"scene\":{\"xaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2},\"yaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2},\"zaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2}},\"shapedefaults\":{\"line\":{\"color\":\"#2a3f5f\"}},\"annotationdefaults\":{\"arrowcolor\":\"#2a3f5f\",\"arrowhead\":0,\"arrowwidth\":1},\"geo\":{\"bgcolor\":\"white\",\"landcolor\":\"#E5ECF6\",\"subunitcolor\":\"white\",\"showland\":true,\"showlakes\":true,\"lakecolor\":\"white\"},\"title\":{\"x\":0.05},\"mapbox\":{\"style\":\"light\"}}},\"legend\":{\"tracegroupgap\":0},\"title\":{\"text\":\"STAR Alignment Mapping Statistics\"}},                        {\"responsive\": true}                    ).then(function(){\n",
              "                            \n",
              "var gd = document.getElementById('69b9930d-8c55-4bcc-8c35-554b9a594fa8');\n",
              "var x = new MutationObserver(function (mutations, observer) {{\n",
              "        var display = window.getComputedStyle(gd).display;\n",
              "        if (!display || display === 'none') {{\n",
              "            console.log([gd, 'removed!']);\n",
              "            Plotly.purge(gd);\n",
              "            observer.disconnect();\n",
              "        }}\n",
              "}});\n",
              "\n",
              "// Listen for the removal of the full notebook cells\n",
              "var notebookContainer = gd.closest('#notebook-container');\n",
              "if (notebookContainer) {{\n",
              "    x.observe(notebookContainer, {childList: true});\n",
              "}}\n",
              "\n",
              "// Listen for the clearing of the current output cell\n",
              "var outputEl = gd.closest('.output');\n",
              "if (outputEl) {{\n",
              "    x.observe(outputEl, {childList: true});\n",
              "}}\n",
              "\n",
              "                        })                };                            </script>        </div>\n",
              "</body>\n",
              "</html>"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "c57df01c"
      },
      "source": [
        "## Display plot\n",
        "\n",
        "### Subtask:\n",
        "Display the generated Plotly plot in the notebook.\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0e016f22"
      },
      "source": [
        "**Reasoning**:\n",
        "Display the generated Plotly plot in the notebook.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 542
        },
        "id": "36b32441",
        "outputId": "b2629690-11b7-482d-a57f-65b81199e0a2"
      },
      "source": [
        "fig.show()"
      ],
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<html>\n",
              "<head><meta charset=\"utf-8\" /></head>\n",
              "<body>\n",
              "    <div>            <script src=\"https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_SVG\"></script><script type=\"text/javascript\">if (window.MathJax && window.MathJax.Hub && window.MathJax.Hub.Config) {window.MathJax.Hub.Config({SVG: {font: \"STIX-Web\"}});}</script>                <script type=\"text/javascript\">window.PlotlyConfig = {MathJaxConfig: 'local'};</script>\n",
              "        <script charset=\"utf-8\" src=\"https://cdn.plot.ly/plotly-2.35.2.min.js\"></script>                <div id=\"33abe2dc-514e-4d88-a9b8-fac187ec6b5d\" class=\"plotly-graph-div\" style=\"height:525px; width:100%;\"></div>            <script type=\"text/javascript\">                                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById(\"33abe2dc-514e-4d88-a9b8-fac187ec6b5d\")) {                    Plotly.newPlot(                        \"33abe2dc-514e-4d88-a9b8-fac187ec6b5d\",                        [{\"domain\":{\"x\":[0.0,1.0],\"y\":[0.0,1.0]},\"hovertemplate\":\"Category=%{label}\\u003cbr\\u003ePercentage=%{value}\\u003cextra\\u003e\\u003c\\u002fextra\\u003e\",\"labels\":[\"Uniquely mapped reads %\",\"% of reads mapped to multiple loci\",\"% of reads mapped to too many loci\",\"% of reads unmapped: too many mismatches\",\"% of reads unmapped: too short\",\"% of reads unmapped: other\"],\"legendgroup\":\"\",\"name\":\"\",\"showlegend\":true,\"values\":[8.76,1.66,0.08,0.0,89.38,0.12],\"type\":\"pie\"}],                        {\"template\":{\"data\":{\"histogram2dcontour\":[{\"type\":\"histogram2dcontour\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"choropleth\":[{\"type\":\"choropleth\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"histogram2d\":[{\"type\":\"histogram2d\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"heatmap\":[{\"type\":\"heatmap\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"heatmapgl\":[{\"type\":\"heatmapgl\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"contourcarpet\":[{\"type\":\"contourcarpet\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"contour\":[{\"type\":\"contour\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"surface\":[{\"type\":\"surface\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"mesh3d\":[{\"type\":\"mesh3d\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"scatter\":[{\"fillpattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2},\"type\":\"scatter\"}],\"parcoords\":[{\"type\":\"parcoords\",\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterpolargl\":[{\"type\":\"scatterpolargl\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"bar\":[{\"error_x\":{\"color\":\"#2a3f5f\"},\"error_y\":{\"color\":\"#2a3f5f\"},\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"bar\"}],\"scattergeo\":[{\"type\":\"scattergeo\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterpolar\":[{\"type\":\"scatterpolar\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"histogram\":[{\"marker\":{\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"histogram\"}],\"scattergl\":[{\"type\":\"scattergl\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatter3d\":[{\"type\":\"scatter3d\",\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scattermapbox\":[{\"type\":\"scattermapbox\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterternary\":[{\"type\":\"scatterternary\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scattercarpet\":[{\"type\":\"scattercarpet\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"carpet\":[{\"aaxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"baxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"type\":\"carpet\"}],\"table\":[{\"cells\":{\"fill\":{\"color\":\"#EBF0F8\"},\"line\":{\"color\":\"white\"}},\"header\":{\"fill\":{\"color\":\"#C8D4E3\"},\"line\":{\"color\":\"white\"}},\"type\":\"table\"}],\"barpolar\":[{\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"barpolar\"}],\"pie\":[{\"automargin\":true,\"type\":\"pie\"}]},\"layout\":{\"autotypenumbers\":\"strict\",\"colorway\":[\"#636efa\",\"#EF553B\",\"#00cc96\",\"#ab63fa\",\"#FFA15A\",\"#19d3f3\",\"#FF6692\",\"#B6E880\",\"#FF97FF\",\"#FECB52\"],\"font\":{\"color\":\"#2a3f5f\"},\"hovermode\":\"closest\",\"hoverlabel\":{\"align\":\"left\"},\"paper_bgcolor\":\"white\",\"plot_bgcolor\":\"#E5ECF6\",\"polar\":{\"bgcolor\":\"#E5ECF6\",\"angularaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"radialaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"ternary\":{\"bgcolor\":\"#E5ECF6\",\"aaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"baxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"caxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"coloraxis\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"colorscale\":{\"sequential\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"sequentialminus\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"diverging\":[[0,\"#8e0152\"],[0.1,\"#c51b7d\"],[0.2,\"#de77ae\"],[0.3,\"#f1b6da\"],[0.4,\"#fde0ef\"],[0.5,\"#f7f7f7\"],[0.6,\"#e6f5d0\"],[0.7,\"#b8e186\"],[0.8,\"#7fbc41\"],[0.9,\"#4d9221\"],[1,\"#276419\"]]},\"xaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"automargin\":true,\"zerolinewidth\":2},\"yaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"automargin\":true,\"zerolinewidth\":2},\"scene\":{\"xaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2},\"yaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2},\"zaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2}},\"shapedefaults\":{\"line\":{\"color\":\"#2a3f5f\"}},\"annotationdefaults\":{\"arrowcolor\":\"#2a3f5f\",\"arrowhead\":0,\"arrowwidth\":1},\"geo\":{\"bgcolor\":\"white\",\"landcolor\":\"#E5ECF6\",\"subunitcolor\":\"white\",\"showland\":true,\"showlakes\":true,\"lakecolor\":\"white\"},\"title\":{\"x\":0.05},\"mapbox\":{\"style\":\"light\"}}},\"legend\":{\"tracegroupgap\":0},\"title\":{\"text\":\"STAR Alignment Mapping Statistics\"}},                        {\"responsive\": true}                    ).then(function(){\n",
              "                            \n",
              "var gd = document.getElementById('33abe2dc-514e-4d88-a9b8-fac187ec6b5d');\n",
              "var x = new MutationObserver(function (mutations, observer) {{\n",
              "        var display = window.getComputedStyle(gd).display;\n",
              "        if (!display || display === 'none') {{\n",
              "            console.log([gd, 'removed!']);\n",
              "            Plotly.purge(gd);\n",
              "            observer.disconnect();\n",
              "        }}\n",
              "}});\n",
              "\n",
              "// Listen for the removal of the full notebook cells\n",
              "var notebookContainer = gd.closest('#notebook-container');\n",
              "if (notebookContainer) {{\n",
              "    x.observe(notebookContainer, {childList: true});\n",
              "}}\n",
              "\n",
              "// Listen for the clearing of the current output cell\n",
              "var outputEl = gd.closest('.output');\n",
              "if (outputEl) {{\n",
              "    x.observe(outputEl, {childList: true});\n",
              "}}\n",
              "\n",
              "                        })                };                            </script>        </div>\n",
              "</body>\n",
              "</html>"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "f76a309f"
      },
      "source": [
        "## Summary:\n",
        "\n",
        "### Data Analysis Key Findings\n",
        "\n",
        "*   The STAR alignment log file was successfully parsed to extract mapping statistics.\n",
        "*   The extracted statistics were organized into a pandas DataFrame, including categories like \"Uniquely mapped reads %\", \"% of reads mapped to multiple loci\", and various unmapped categories.\n",
        "*   A Plotly pie chart was generated to visualize the distribution of reads across these mapping categories.\n",
        "\n",
        "### Insights or Next Steps\n",
        "\n",
        "*   The pie chart provides a clear visual overview of the STAR alignment efficiency, highlighting the proportion of reads in different mapping categories.\n",
        "*   Analyze the percentages of unmapped reads to identify potential issues with read quality, reference genome, or alignment parameters.\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "2c820987"
      },
      "source": [
        "Here is a detailed interpretation of the key metrics from the `star_output/SRR11548957_Log.final.out` file:\n",
        "\n",
        "*   **Started job on**: Indicates the date and time when the STAR alignment job began.\n",
        "*   **Started mapping on**: Indicates the date and time when the read mapping process started.\n",
        "*   **Finished on**: Indicates the date and time when the entire STAR alignment job finished.\n",
        "*   **Mapping speed, Million of reads per hour**: This metric shows the processing speed of STAR, indicating how many million reads were aligned per hour. A higher number indicates faster processing.\n",
        "*   **Number of input reads**: The total number of reads provided as input to STAR.\n",
        "*   **Average input read length**: The average length of the input reads.\n",
        "*   **UNIQUE READS:** Section detailing statistics for reads that mapped to only one location in the reference genome.\n",
        "*   **Uniquely mapped reads number**: The total count of reads that mapped uniquely to the genome.\n",
        "*   **Uniquely mapped reads %**: The percentage of total input reads that mapped uniquely. This is a crucial metric for assessing the quality of the alignment. A higher percentage is generally better.\n",
        "*   **Average mapped length**: The average length of the uniquely mapped reads.\n",
        "*   **Number of splices: Total**: The total number of splice junctions detected by STAR.\n",
        "*   **Number of splices: Annotated (sjdb)**: The number of detected splice junctions that match known splice junctions provided in the GTF annotation file.\n",
        "*   **Number of splices: GT/AG**: The number of splice junctions with the canonical GT/AG intron motif. This is the most common type of splice junction.\n",
        "*   **Number of splices: GC/AG**: The number of splice junctions with the GC/AG intron motif, another common type.\n",
        "*   **Number of splices: AT/AC**: The number of splice junctions with the less common AT/AC intron motif.\n",
        "*   **Number of splices: Non-canonical**: The number of splice junctions that do not match any of the canonical or common non-canonical motifs. A high number here might indicate issues with the data or alignment.\n",
        "*   **Mismatch rate per base, %**: The percentage of bases that are mismatched between the reads and the reference genome. A low mismatch rate is desirable.\n",
        "*   **Deletion rate per base**: The rate of deletions per base in the aligned reads compared to the reference.\n",
        "*   **Deletion average length**: The average length of deletions.\n",
        "*   **Insertion rate per base**: The rate of insertions per base in the aligned reads compared to the reference.\n",
        "*   **Insertion average length**: The average length of insertions.\n",
        "*   **MULTI-MAPPING READS:** Section detailing statistics for reads that mapped to more than one location in the reference genome.\n",
        "*   **Number of reads mapped to multiple loci**: The total count of reads that mapped to multiple locations.\n",
        "*   **% of reads mapped to multiple loci**: The percentage of total input reads that mapped to multiple locations.\n",
        "*   **Number of reads mapped to too many loci**: The count of reads that mapped to more locations than the maximum allowed by STAR's parameters.\n",
        "*   **% of reads mapped to too many loci**: The percentage of total input reads that mapped to too many locations.\n",
        "*   **UNMAPPED READS:** Section detailing statistics for reads that did not map to the reference genome.\n",
        "*   **Number of reads unmapped: too many mismatches**: The count of reads that did not map due to exceeding the allowed number of mismatches.\n",
        "*   **% of reads unmapped: too many mismatches**: The percentage of total input reads that did not map due to too many mismatches.\n",
        "*   **Number of reads unmapped: too short**: The count of reads that did not map because they were too short after trimming or soft clipping.\n",
        "*   **% of reads unmapped: too short**: The percentage of total input reads that did not map because they were too short.\n",
        "*   **Number of reads unmapped: other**: The count of reads that did not map for reasons other than too many mismatches or being too short.\n",
        "*   **% of reads unmapped: other**: The percentage of total input reads that did not map for other reasons.\n",
        "*   **CHIMERIC READS:** Section detailing statistics for chimeric reads (reads that map to two different locations).\n",
        "*   **Number of chimeric reads**: The total count of chimeric reads detected.\n",
        "*   **% of chimeric reads**: The percentage of total input reads that were identified as chimeric."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "4753e21e"
      },
      "source": [
        "Here is a summary of the key findings from the STAR alignment of SRR11548957:\n",
        "\n",
        "*   **Mapping Speed:** The alignment processed reads at a rate of 4.74 million reads per hour.\n",
        "*   **Input Reads:** A total of 1,792,464 reads were input for alignment, with an average length of 150 bases.\n",
        "*   **Uniquely Mapped Reads:** Only 8.76% of the reads (156,947) uniquely mapped to the reference genome (chromosome 22). This is a relatively low unique mapping rate, suggesting either the reads are from a different genome/chromosome, the reference genome is incomplete, or there are issues with the read quality or adapter trimming.\n",
        "*   **Multi-mapping Reads:** 1.66% of reads (29,782) mapped to multiple locations, and a very small percentage (0.08%) mapped to too many loci.\n",
        "*   **Unmapped Reads:** A large majority of reads (89.38%) were unmapped because they were too short. This is a significant percentage and might indicate issues with the quality of the input reads or the chosen alignment parameters. A small percentage (0.12%) were unmapped for other reasons.\n",
        "*   **Splice Junctions:** A total of 12,073 splice junctions were detected, with a good proportion (6,261) being annotated in the provided GTF file. The majority of detected junctions have canonical GT/AG (10,520) and GC/AG (209) motifs, with a smaller number of non-canonical junctions (1337).\n",
        "*   **Mismatch, Deletion, and Insertion Rates:** The mismatch rate per base was 5.16%, which is quite high. The deletion and insertion rates were lower, at 0.11% and 0.17% respectively.\n",
        "\n",
        "**Overall Assessment:** The alignment had a very low unique mapping rate and a high percentage of reads unmapped due to being too short, as well as a high mismatch rate. This suggests potential issues with the input data quality or that the chosen reference (chromosome 22) may not be the most appropriate for these reads. Further investigation into the read quality and the expected origin of the sequencing data would be recommended."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "513b04cd"
      },
      "source": [
        "**Conclusion of the RNA-Seq Alignment Workflow:**\n",
        "\n",
        "This notebook demonstrates a basic RNA-Seq alignment workflow using STAR, focusing on a subset of data (SRR11548957) and a specific reference genome (human chromosome 22). The steps performed were:\n",
        "\n",
        "1.  **Environment Setup:** Installed necessary tools and libraries including `sra-toolkit` for data download and `STAR` for alignment.\n",
        "2.  **Data Download:** Downloaded paired-end FASTQ files for the accession SRR11548957 from the SRA database.\n",
        "3.  **Reference Preparation:** Downloaded and uncompressed the reference genome sequence (chromosome 22) and the corresponding GTF annotation file from Ensembl.\n",
        "4.  **Genome Indexing:** Generated a STAR genome index using the downloaded reference genome and GTF file. This is a crucial step for efficient alignment.\n",
        "5.  **Read Alignment:** Aligned the downloaded FASTQ reads to the generated STAR index, producing a sorted BAM file and gene counts.\n",
        "6.  **Output Inspection and Download:** Inspected key output files (`Aligned.sortedByCoord.out.bam`, `ReadsPerGene.out.tab`, and `Log.final.out`) and provided code to download them.\n",
        "\n",
        "**Summary of Findings:**\n",
        "\n",
        "Based on the `Log.final.out` analysis, while the alignment successfully ran, the unique mapping rate was low (8.76%), and a high percentage of reads (89.38%) were unmapped due to being too short. The mismatch rate was also relatively high (5.16%). This suggests that the reads might not be primarily from chromosome 22 or that there are quality issues with the sequencing data.\n",
        "\n",
        "**Next Steps:**\n",
        "\n",
        "For a complete RNA-Seq analysis, the next steps would typically involve:\n",
        "\n",
        "*   More in-depth quality control of the raw reads (e.g., using FastQC and MultiQC before alignment).\n",
        "*   Considering aligning to the full human genome for a more comprehensive analysis.\n",
        "*   Performing differential gene expression analysis using the gene counts obtained from STAR and appropriate statistical software (e.g., DESeq2, edgeR).\n",
        "*   Further downstream analysis and visualization of the results."
      ]
    }
  ]
}